File: windows/QuotaArc/Model/Fixtures.cs
using QuotaArc.Providers;

namespace QuotaArc.Model;

internal static class Fixtures
{
    public static List<ProviderSnapshot> Snapshots(DateTime? now = null)
    {
        var n = now ?? DateTime.Now;
        var sessionReset = n.AddMinutes(51);
        var midnight = n.Date.AddDays(1);
        return
        [
            new ProviderSnapshot(
                "claude", "Claude", ProviderGlyph.Claude, Fidelity.Official,
                new ProviderStatus.Ok(),
                [
                    new LimitWindow("session", "Current session", 0.73, ResetsAt: sessionReset),
                    new LimitWindow("weekly_all", "All models", 0.07, ResetsAt: midnight)
                ], "session"),
            new ProviderSnapshot(
                "cursor", "Cursor", ProviderGlyph.Cursor, Fidelity.Official,
                new ProviderStatus.Ok(),
                [new LimitWindow("included", "Included usage", 0.52, ResetsAt: midnight)],
                "included"),
            new ProviderSnapshot(
                "codex", "Codex", ProviderGlyph.Openai, Fidelity.Official,
                new ProviderStatus.Ok(),
                [
                    new LimitWindow("primary", "5h limit", 0.21, ResetsAt: n.AddHours(3)),
                    new LimitWindow("secondary", "Weekly limit", 0.34, ResetsAt: n.AddDays(4))
                ], "primary"),
            new ProviderSnapshot(
                "gemini", "Antigravity", ProviderGlyph.Antigravity, Fidelity.Official,
                new ProviderStatus.Ok(),
                [new LimitWindow("requests", "Requests today", Used: 18)])
        ];
    }
}

File: Sources/Model/Fixtures.swift
import Foundation

/// The three providers from the design frame, at the levels it shows.
/// These stand in until the adapters in M4 land.
enum Fixtures {
    static func snapshots(now: Date = Date(), calendar: Calendar = .current) -> [ProviderSnapshot] {
        let sessionReset = now.addingTimeInterval(51 * 60)
        let midnight = calendar.startOfDay(for: now.addingTimeInterval(24 * 60 * 60))

        return [
            ProviderSnapshot(
                id: "claude",
                displayName: "Claude",
                glyph: .claude,
                fidelity: .derived,
                status: .ok,
                windows: [
                    LimitWindow(id: "claude.session", label: "Current session",
                                usedFraction: 0.73, resetsAt: sessionReset),
                    LimitWindow(id: "claude.all", label: "All models",
                                usedFraction: 0.07, resetsAt: midnight)
                ]
            ),
            ProviderSnapshot(
                id: "openai",
                displayName: "OpenAI",
                glyph: .openai,
                fidelity: .manual,
                status: .ok,
                windows: [
                    LimitWindow(id: "openai.session", label: "Current session",
                                usedFraction: 0.21, resetsAt: now.addingTimeInterval(3 * 60 * 60))
                ]
            ),
            ProviderSnapshot(
                id: "third",
                displayName: "Perplexity",
                glyph: .third,
                fidelity: .manual,
                status: .ok,
                windows: [
                    LimitWindow(id: "third.daily", label: "Daily quota",
                                usedFraction: 0.52, resetsAt: midnight)
                ]
            )
        ]
    }
}
